getCurrentTime() and setCurrentTime() for AudioSourceComponent

nakata0705 opened the pull request:
These two simple interface functions let users to set/get currentTime of audio from AudioSourceComponent. Checked against Chrome 47, FireFox 42 and IE 11.

willeastcott commented on src/framework/components/audiosource/audiosource_component.js:
```diff
@@ -101,6 +109,35 @@ pc.extend(pc, function () {
+         * @name pc.AudioSourceComponent#getCurrentTime
+         * @description Get the current time of the playback.
+         */
+        getCurrentTime: function () {
```
I think the API here should match the animation component:
https://github.com/playcanvas/engine/blob/master/src/framework/components/animation/animation_component.js#L268
In other words, rather than functions, a property would be better.

nakata0705 commented:
Thank you for your review Will. That makes sense.

I updated my branch and implemented properties as you proposed. Now AudioSourceComponent has currentTime and duration properties.

willeastcott commented:
No problem, Nakata-san! :) A couple of things:
1. Please complete the Contributor License Agreement [here](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform) so we can merge your PR.
2. Please test your engine changes against your project. To do this, build the engine (run 'build.py' in the build folder), run a local web server (in a command prompt, navigate to the build\output folder and run a local server. python -m SimpleHTTPServer 51000). Then add '?use_local_engine=true' to the URL of your launched app running. Then your local engine will be used instead of the one on code.playcanvas.com

nakata0705 commented:
Hello Will,

1. I believe I already signed to Contributor License Agreement for the
translation team. I included the github account when I submit. So this
should be already completed.

2. This is an easy and handy method! I'm encountering an unexpected
behavior.

I have been testing my API change with a modified spin cube example in my
local disk. The new interface has been working well during this debug. So I
sent a pull request.

FYI: My local debug code is in...
https://github.com/nakata0705/engine/tree/debug/examples/audio_test
https://github.com/nakata0705/engine/tree/debug/examples/assets

But when I tried your method with my work-in-progress project in
PlayCanvas, I noticed that the music just stops after changing the
currentTime from the slide bar. I chased the issue and I realized that
there are two extra channel.pause() call from unknown location that didn't
happen when I run the modified engine code at local. I'm currently kind of
in stuck. If you would think of any cause, that would be great!

FYI: My work-in-progress that uses my updated engine is in...
https://playcanvas.com/project/368712/overview/work-in-progress

This is the expected 1st channel.pause() call from the new interface.
channel.unpause() is called right after this as I coded in the new set()
for currentTime.
[image: Inline image 5]

This is the unexpected 2nd channel.pause() call from somewhere. Somehow the
debugger doesn't display the call stack.

[image: Inline image 6]

This is the unexpected 3rd channel.pause() call.
[image: Inline image 7]

If I hit "step out from this function call" from the 2nd or 3rd
channel.pause() call, it goes to here. Currently I don't fully understand
what's going on in this situation.
[image: Inline image 10]

On Sun, Dec 13, 2015 at 7:31 AM, Will Eastcott <notifications@github.com>
wrote:

> No problem, Nakata-san! :) A couple of things:
> 1. Please complete the Contributor License Agreement here
> <https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform>
> so we can merge your PR.
> 2. Please test your engine changes against your project. To do this, build
> the engine (run 'build.py' in the build folder), run a local web server (in
> a command prompt, navigate to the build\output folder and run a local
> server. python -m SimpleHTTPServer 51000). Then add
> '?use_local_engine=true' to the URL of your launched app running. Then your
> local engine will be used instead of the one on code.playcanvas.com
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/playcanvas/engine/pull/473#issuecomment-164268185>.
>

willeastcott commented:
Any ideas, @vkalpias?

daredevildave commented:
Thanks for the PR, but this is no longer required. As it is covered in the new Sound Component API.

http://developer.playcanvas.com/en/api/pc.SoundComponent.html
